src/Help/TicketSystem/TicketMessageRepository.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EtoA\Help\TicketSystem;
+
+use Doctrine\DBAL\Connection;
+use EtoA\Admin\AdminUserRepository;
+use EtoA\Core\AbstractRepository;
+use EtoA\User\UserRepository;
+
+class TicketMessageRepository extends AbstractRepository
+{
+    private AdminUserRepository $adminUserRepo;
+    private UserRepository $userRepo;
+
+    public function __construct(
+        Connection $connection,
+        AdminUserRepository $adminUserRepo,
+        UserRepository $userRepo
+    ) {
+        parent::__construct($connection);
+        $this->adminUserRepo = $adminUserRepo;
+        $this->userRepo = $userRepo;
+    }
+
+    public function count(int $ticketId): int
+    {
+        return (int) $this->createQueryBuilder()
+            ->select("COUNT(*)")
+            ->from('ticket_msg')
+            ->where('ticket_id = :ticket_id')
+            ->setParameter('ticket_id', $ticketId)
+            ->execute()
+            ->fetchOne();
+    }
+
+    public function find(int $id): ?TicketMessage
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('ticket_msg')
+            ->where('id = :id')
+            ->setParameter('id', $id)
+            ->execute()
+            ->fetchAssociative();
+        return $data ? $this->createObject($data) : null;
+    }
+
+    public function findByTicket(int $ticketId): array
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('ticket_msg')
+            ->where('ticket_id = :ticket_id')
+            ->orderBy('timestamp', 'ASC')
+            ->setParameter('ticket_id', $ticketId)
+            ->execute()
+            ->fetchAllAssociative();
+        return collect($data)
+            ->map(fn ($arr) => $this->createObject($arr))
+            ->toArray();
+    }
+
+    public function findLastMessageForTicket(int $ticketId): ?TicketMessage
+    {
+        $data = $this->createQueryBuilder()
+            ->select("*")
+            ->from('ticket_msg')
+            ->where('ticket_id = :ticket_id')
+            ->orderBy('timestamp', 'DESC')
+            ->addOrderBy('id', 'DESC')
+            ->setParameter('ticket_id', $ticketId)
+            ->execute()
+            ->fetchAssociative();
+        return $data ? $this->createObject($data) : null;
+    }
+
+    private function createObject(array $data): TicketMessage
+    {
+        $message = new TicketMessage();
+
+        $message->id = (int) $data['id'];
+        $message->ticketId = (int) $data['ticket_id'];
+        $message->userId = (int) $data['user_id'];
+        $message->adminId = (int) $data['admin_id'];
+        $message->timestamp = (int) $data['timestamp'];
+        $message->message = $data['message'];
+
+        return $message;
+    }
+
+	public function create($data): TicketMessage

```

Comment:
instead of data would pass in the TicketMessage

Reply:
Done c3e46cd